src/BackupManager/Factory/DatabaseFactory.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace DigipolisGent\Robo\Task\Deploy\BackupManager\Factory;
+
+use BackupManager\Config\Config;
+use BackupManager\Databases\DatabaseProvider;
+use BackupManager\Databases\PostgresqlDatabase;
+use DigipolisGent\Robo\Task\Deploy\BackupManager\Databases\MysqlDatabase;
+
+class DatabaseFactory implements DatabaseFactoryInterface
+{
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function create($dbConfig)
+    {
+        $dbConfigObj = is_array($dbConfig)
+            ? new Config($dbConfig)
+            : Config::fromPhpFile($dbConfig);
+
+        // Add all default databases.
+        $databaseProvider = new DatabaseProvider($dbConfigObj);
+        $databaseProvider->add(new MysqlDatabase());
+        $databaseProvider->add(new PostgresqlDatabase());
```

Comment:
You define your db type in config. The rest of the config options are determined by the type of db you specified. The library determines which database to use based on the passed config and just passes the config to the right class. So besides the 'type' parameter in the config which is required in every case, all other config items can be totally different. So in short: Options differ from type to type.